BlueprintUI/Sources/Internal/LayoutResultNode.swift

```diff
@@ -18,8 +18,34 @@ extension Element {
     }
 
     func layout(frame: CGRect, environment: Environment, layoutMode: LayoutMode) -> LayoutResultNode {
-        // TODO: switch on layoutMode
-        layout(layoutAttributes: LayoutAttributes(frame: frame), environment: environment)
+        switch layoutMode {
+        case .standard:
+            return layout(layoutAttributes: LayoutAttributes(frame: frame), environment: environment)
+        case .singlePass:
+            return singlePassLayout(
+                proposal: .init(frame.size),
+                context: SPLayoutContext(
+                    attributes: LayoutAttributes(frame: frame),
+                    environment: environment,
+                    // TODO: Hoist up?
```

Comment:
Same here regarding potentially doing this before we get too deep into performance optimizations.